system-addon/content-src/lib/init-store.js

```diff
@@ -76,6 +77,28 @@ const rehydrationMiddleware = store => next => action => {
   return next(action);
 };
 
+// This middleware queues up all the EARLY_QUEUED_ACTIONS until it receives
+// the first action from main. Note that NEW_TAB_INITIAL_STATE is guaranteed
+// to be sent from main
```

Comment:
Switch this to a `/* */` comment block, and explain things a bit better probably without the "NEW_TAB_INITIAL_STATE" mention. Probably something including "because any messages for main before receiving from main might be lost. Any actions allowed early are accepted to be unreliable, i.e., can handle re-sending or be ignored."